File: SwaggerApp/Controllers/TestController.cs
using System;
using System.Data;
using System.Linq;
using System.Net;
using System.Net.Http;
using System.Security.Claims;
using System.Text;
using System.Web.Http;
using System.Web.Http.Description;
using Microsoft.VisualBasic;
using Newtonsoft.Json;
using Swagger.Net.Annotations;

namespace SwaggerApp.Controllers
{
    /// <summary>
    /// Metodos de Prueba
    /// </summary>
    [SwaggerResponse(HttpStatusCode.Unauthorized, "Ha sido negada esta solicitud por falta de autorizacion.")]
    [SwaggerResponse(HttpStatusCode.Forbidden, "No tiene permisos para esta solicitud.")]
    [SwaggerResponse(HttpStatusCode.BadRequest, "Error en solicitud.")]
    [SwaggerResponse(HttpStatusCode.InternalServerError, "Error de aplicacion interno.")]
    [Authorize]
    public class TestController : ApiController
    {
        ///// <summary>
        ///// Consulta todos los datos
        ///// </summary>
        ///// <response code="200" cref="GetResponse">Operacion exitosa.</response>
        //[Route("testget")]
        //[ResponseType(typeof(GetResponse))]
        //[HttpGet]
        //public IHttpActionResult GetAll()
        //{
        //    HttpResponseMessage response;
        //    response = Request.CreateResponse(HttpStatusCode.OK);

        //    string jsonR = "{\"Nombre\":\"roberto\",\"Mensaje\":\"Respuesta Ok\"}";

        //    response.Content = new StringContent(jsonR, Encoding.UTF8, "application/json");
        //    response.Headers.Location = new Uri(Request.RequestUri.ToString());
        //    return base.ResponseMessage(response);
        //}

        ///// <summary>
        ///// Consulta un solo dato
        ///// </summary>
        ///// <response code="200" cref="GetResponse">Operacion exitosa.</response>
        ///// <param name="id">Numero de Documento</param>
        //[Route("testget/{id}")]
        //[ResponseType(typeof(GetResponse))]
        //[HttpGet]
        //public IHttpActionResult GetAll(string id)
        //{
        //    HttpResponseMessage response;
        //    response = Request.CreateResponse(HttpStatusCode.OK);

        //    string jsonR = "{\"Nombre\":\"roberto\",\"Mensaje\":\"Respuesta Ok\"}";

        //    response.Content = new StringContent(jsonR, Encoding.UTF8, "application/json");
        //    response.Headers.Location = new Uri(Request.RequestUri.ToString());
        //    return base.ResponseMessage(response);
        //}

        /// <summary>
        /// Consulta un solo dato mas parametros
        /// </summary>
        /// <response code="200" cref="GetResponse">Operacion exitosa.</response>
        /// <param name="id">Numero de documento</param>
        /// <param name="sort">1 = Ordenar por nombre (opcional)</param>
        [Route("get/{id}")]
        [ResponseType(typeof(GetResponse))]
        [HttpGet]
        public IHttpActionResult Get(int id = 0, string sort = "0")
        {
            HttpResponseMessage response = Request.CreateResponse(HttpStatusCode.OK);

            //const string jsonR = "{\"id\":\" + id + "\",\"Mensaje\":\"Respuesta Ok\"}";

            var resp = new GetResponse()
            {
                Nombre = "Roberto",
                Numero = id,
                Mensaje = "Ya Existe."
            };

            string jsonR = JsonConvert.SerializeObject(resp);

            response.Content = new StringContent(jsonR, Encoding.UTF8, "application/json");
            response.Headers.Location = new Uri(Request.RequestUri.ToString() + "/" + id);
            return base.ResponseMessage(response);
        }

        /// <summary>
        /// Alta de Documento.
        /// </summary>
        /// <response code="201" cref="PostResponse">Alta de numero exitosa.</response>
        /// <response code="303" cref="PostResponse">Numero ya se encontraba en el sistema (identificador en Location header, swagger redirecciona automaticamente.).</response>
        /// <param name="value">Datos del certificado.</param>
        [Route("post")]
        [ResponseType(typeof(PostResponse))]
        [HttpPost]
        public IHttpActionResult Post([FromBody] PostData value)
        {
            if (value.Numero == 351)
            {
                var resp = new PostResponse()
                {
                    Nombre = value.Nombre,
                    Numero = value.Numero,
                    Mensaje = "Ya Existe."
                };

                string jsonR = JsonConvert.SerializeObject(resp);
                HttpResponseMessage responseG = Request.CreateResponse(HttpStatusCode.SeeOther);
                responseG.Content = new StringContent(jsonR, Encoding.UTF8, "application/json");
                responseG.Headers.Location = new Uri(Request.RequestUri.ToString() + "/" + value.Numero);
                return base.ResponseMessage(responseG);
            }
            else
            {
                var resp = new PostResponse()
                {
                    Nombre = value.Nombre,
                    Numero = value.Numero,
                    Mensaje = "Alta Correcta."
                };

                string jsonR = JsonConvert.SerializeObject(resp);

                HttpResponseMessage responseG = Request.CreateResponse(HttpStatusCode.Created);
                responseG.Content = new StringContent(jsonR, Encoding.UTF8, "application/json");
                responseG.Headers.Location = new Uri(Request.RequestUri.ToString() + "/" + value.Numero);
                return base.ResponseMessage(responseG);
            }
        }

        /// <summary>
        /// Modificacion de Documento.
        /// </summary>
        /// <response code="200" cref="PostResponse">Modificacion de numero exitosa.</response>
        /// <param name="id">Numero a modificar</param>
        /// <param name="value">Datos del certificado.</param>
        [Route("put/{id}")]
        [ResponseType(typeof(PostResponse))]
        [HttpPut]
        public IHttpActionResult Put(int id, [FromBody] PutData value)
        {
            var resp = new PostResponse()
            {
                Nombre = value.Nombre,
                Numero = id,
                Mensaje = "Modificacion Correcta."
            };

            string jsonR = JsonConvert.SerializeObject(resp);

            HttpResponseMessage responseG = Request.CreateResponse(HttpStatusCode.Created);
            responseG.Content = new StringContent(jsonR, Encoding.UTF8, "application/json");
            responseG.Headers.Location = new Uri(Request.RequestUri.ToString() + "/" + id);
            return base.ResponseMessage(responseG);
        }

        /// <summary>
        /// Elimina Documento.
        /// </summary>
        /// <response code="200">Operacion exitosa.</response>
        /// <param name="id">Numero de documento</param>
        [Route("delete/{id}")]
        [HttpDelete]
        public IHttpActionResult Delete(int id)
        {
            HttpResponseMessage responseG = Request.CreateResponse(HttpStatusCode.OK);
            return base.ResponseMessage(responseG);
        }

        /// <summary>
        /// Actualizacion de Documento.
        /// </summary>
        /// <response code="200" cref="PostResponse">Modificacion de numero exitosa.</response>
        /// <param name="id">Numero a modificar</param>
        /// <param name="value">Datos del certificado.</param>
        [Route("patch/{id}")]
        [ResponseType(typeof(PostResponse))]
        [HttpPatch]
        public IHttpActionResult Patch(int id, [FromBody] PutData value)
        {
            var resp = new PostResponse()
            {
                Nombre = value.Nombre,
                Numero = id,
                Mensaje = "Modificacion Correcta."
            };

            string jsonR = JsonConvert.SerializeObject(resp);

            HttpResponseMessage responseG = Request.CreateResponse(HttpStatusCode.Created);
            responseG.Content = new StringContent(jsonR, Encoding.UTF8, "application/json");
            responseG.Headers.Location = new Uri(Request.RequestUri.ToString() + "/" + id);
            return base.ResponseMessage(responseG);
        }
    }
}

File: SwaggerApp/Controllers/TestController.vb
Imports System
Imports System.Net
Imports System.Net.Http
Imports System.Text
Imports System.Web.Http
Imports Newtonsoft.Json
Imports Swagger.Net.Annotations

Namespace SwaggerApp.Controllers
    ''' <summary>
    ''' Metodos de Prueba
    ''' </summary>
    <SwaggerResponse(HttpStatusCode.Unauthorized, "Ha sido negada esta solicitud por falta de autorizacion.")>
    <SwaggerResponse(HttpStatusCode.Forbidden, "No tiene permisos para esta solicitud.")>
    <SwaggerResponse(HttpStatusCode.BadRequest, "Error en solicitud.")>
    <SwaggerResponse(HttpStatusCode.InternalServerError, "Error de aplicacion interno.")>
    <SwaggerApp.AuthorizeAttribute>
    Public Class TestController
        Inherits ApiController
        '''' <summary>
        '''' Consulta todos los datos
        '''' </summary>
        '''' <response code="200" cref="GetResponse">Operacion exitosa.</response>
        '[Route("testget")]
        '[ResponseType(typeof(GetResponse))]
        '[HttpGet]
        'public IHttpActionResult GetAll()
        '{
        '    HttpResponseMessage response;
        '    response = Request.CreateResponse(HttpStatusCode.OK);

        '    string jsonR = "{\"Nombre\":\"roberto\",\"Mensaje\":\"Respuesta Ok\"}";

        '    response.Content = new StringContent(jsonR, Encoding.UTF8, "application/json");
        '    response.Headers.Location = new Uri(Request.RequestUri.ToString());
        '    return base.ResponseMessage(response);
        '}

        '''' <summary>
        '''' Consulta un solo dato
        '''' </summary>
        '''' <response code="200" cref="GetResponse">Operacion exitosa.</response>
        '''' <param name="id">Numero de Documento</param>
        '[Route("testget/{id}")]
        '[ResponseType(typeof(GetResponse))]
        '[HttpGet]
        'public IHttpActionResult GetAll(string id)
        '{
        '    HttpResponseMessage response;
        '    response = Request.CreateResponse(HttpStatusCode.OK);

        '    string jsonR = "{\"Nombre\":\"roberto\",\"Mensaje\":\"Respuesta Ok\"}";

        '    response.Content = new StringContent(jsonR, Encoding.UTF8, "application/json");
        '    response.Headers.Location = new Uri(Request.RequestUri.ToString());
        '    return base.ResponseMessage(response);
        '}

        ''' <summary>
        ''' Consulta un solo dato mas parametros
        ''' </summary>
        ''' <responsecode="200"cref="GetResponse">Operacion exitosa.</response>
        ''' <paramname="id">Numero de documento</param>
        ''' <paramname="sort">1 = Ordenar por nombre (opcional)</param>
        <Route("get/{id}")>
        <Description.ResponseTypeAttribute(GetType(SwaggerApp.GetResponse))>
        <HttpGet>
        Public Function [Get](ByVal Optional id As Integer = 0, ByVal Optional sort As String = "0") As IHttpActionResult
            Dim response = Request.CreateResponse(HttpStatusCode.OK)

            'const string jsonR = "{\"id\":\" + id + "\",\"Mensaje\":\"Respuesta Ok\"}";

            Dim resp = New SwaggerApp.GetResponse() With {
                .Nombre = "Roberto",
                .Numero = id,
                .Mensaje = "Ya Existe."
            }
            Dim jsonR = JsonConvert.SerializeObject(resp)
            response.Content = New StringContent(jsonR, Encoding.UTF8, "application/json")
            response.Headers.Location = New Uri(Request.RequestUri.ToString() & "/" & id)
            Return MyBase.ResponseMessage(response)
        End Function

        ''' <summary>
        ''' Alta de Documento.
        ''' </summary>
        ''' <responsecode="201"cref="PostResponse">Alta de numero exitosa.</response>
        ''' <responsecode="303"cref="PostResponse">Numero ya se encontraba en el sistema (identificador en Location header, swagger redirecciona automaticamente.).</response>
        ''' <paramname="value">Datos del certificado.</param>
        <Route("post")>
        <Description.ResponseTypeAttribute(GetType(SwaggerApp.PostResponse))>
        <HttpPost>
        Public Function Post(
        <FromBody> ByVal value As SwaggerApp.PostData) As IHttpActionResult
            If value.Numero = 351 Then
                Dim resp = New SwaggerApp.PostResponse() With {
                    .Nombre = value.Nombre,
                    .Numero = value.Numero,
                    .Mensaje = "Ya Existe."
                }
                Dim jsonR = JsonConvert.SerializeObject(resp)
                Dim responseG = Request.CreateResponse(HttpStatusCode.SeeOther)
                responseG.Content = New StringContent(jsonR, Encoding.UTF8, "application/json")
                responseG.Headers.Location = New Uri(Request.RequestUri.ToString() & "/" & value.Numero)
                Return MyBase.ResponseMessage(responseG)
            Else
                Dim resp = New SwaggerApp.PostResponse() With {
                    .Nombre = value.Nombre,
                    .Numero = value.Numero,
                    .Mensaje = "Alta Correcta."
                }
                Dim jsonR = JsonConvert.SerializeObject(resp)
                Dim responseG = Request.CreateResponse(HttpStatusCode.Created)
                responseG.Content = New StringContent(jsonR, Encoding.UTF8, "application/json")
                responseG.Headers.Location = New Uri(Request.RequestUri.ToString() & "/" & value.Numero)
                Return MyBase.ResponseMessage(responseG)
            End If
        End Function

        ''' <summary>
        ''' Modificacion de Documento.
        ''' </summary>
        ''' <responsecode="200"cref="PostResponse">Modificacion de numero exitosa.</response>
        ''' <paramname="id">Numero a modificar</param>
        ''' <paramname="value">Datos del certificado.</param>
        <Route("put/{id}")>
        <Description.ResponseTypeAttribute(GetType(SwaggerApp.PostResponse))>
        <HttpPut>
        Public Function Put(ByVal id As Integer,
        <FromBody> ByVal value As SwaggerApp.PutData) As IHttpActionResult
            Dim resp = New SwaggerApp.PostResponse() With {
                .Nombre = value.Nombre,
                .Numero = id,
                .Mensaje = "Modificacion Correcta."
            }
            Dim jsonR = JsonConvert.SerializeObject(resp)
            Dim responseG = Request.CreateResponse(HttpStatusCode.Created)
            responseG.Content = New StringContent(jsonR, Encoding.UTF8, "application/json")
            responseG.Headers.Location = New Uri(Request.RequestUri.ToString() & "/" & id)
            Return MyBase.ResponseMessage(responseG)
        End Function

        ''' <summary>
        ''' Elimina Documento.
        ''' </summary>
        ''' <responsecode="200">Operacion exitosa.</response>
        ''' <paramname="id">Numero de documento</param>
        <Route("delete/{id}")>
        <HttpDelete>
        Public Function Delete(ByVal id As Integer) As IHttpActionResult
            Dim responseG = Request.CreateResponse(HttpStatusCode.OK)
            Return MyBase.ResponseMessage(responseG)
        End Function

        ''' <summary>
        ''' Actualizacion de Documento.
        ''' </summary>
        ''' <responsecode="200"cref="PostResponse">Modificacion de numero exitosa.</response>
        ''' <paramname="id">Numero a modificar</param>
        ''' <paramname="value">Datos del certificado.</param>
        <Route("patch/{id}")>
        <Description.ResponseTypeAttribute(GetType(SwaggerApp.PostResponse))>
        <HttpPatch>
        Public Function Patch(ByVal id As Integer,
        <FromBody> ByVal value As SwaggerApp.PutData) As IHttpActionResult
            Dim resp = New SwaggerApp.PostResponse() With {
                .Nombre = value.Nombre,
                .Numero = id,
                .Mensaje = "Modificacion Correcta."
            }
            Dim jsonR = JsonConvert.SerializeObject(resp)
            Dim responseG = Request.CreateResponse(HttpStatusCode.Created)
            responseG.Content = New StringContent(jsonR, Encoding.UTF8, "application/json")
            responseG.Headers.Location = New Uri(Request.RequestUri.ToString() & "/" & id)
            Return MyBase.ResponseMessage(responseG)
        End Function
    End Class
End Namespace
